packages/common/src/apiutil/RequestQueue.ts

```diff
@@ -0,0 +1,95 @@
+import { requestLogger } from "./logging.js";
+
+/**
+ * The RequestQueue keeps a queue of all requests done to the server and
+ * executes them one by one in the order they are put into the queue (FIFO).
+ */
+export class RequestQueue {
+    name: string
+    baseQueue: BaseQueue<Job> = new BaseQueue<Job>()
+    
+    constructor(name: string) {
+        this.name = name
+    }
+
+    /**
+     * Add _job_ to the queue.
+     * @param job
+     */
+    add(job: Job): void {
+        if (this.baseQueue.isEmpty() && !this.running) {
+            this.baseQueue.enqueue(job)
+            this.run()
+        } else {
+            this.baseQueue.enqueue(job)
+        }
+    }
+
+    /**
+     * Start executing requests that are on the queue.
+     */
+    async run() {
+        // requestLogger.info("RequestQueue.run start")
+        this.running = true
+        try {
+            while (!this.baseQueue.isEmpty()) {
+                const job = this.baseQueue.dequeue()
+                if (job !== undefined) {
+                    // requestLogger.info("RequestQueue.run job: " + job.name)
```

Comment:
Commented out code

Reply:
Removed (changed this class quite a lot).